AbcSim.h

```diff
@@ -40,16 +40,16 @@ namespace ABC {
 // In general, the goal is that those implementations should *not* be "stateful". That is, they should not have any internal state which
 // is changed when they are used.
 struct AbcSimFun {
-    virtual vector<ABC::float_type> operator()(
-      vector<ABC::float_type> pars, const unsigned long int seed, const unsigned long int serial, const ABC::MPI_par* _mp
+    virtual vector<float_type> operator()(
+      vector<float_type> pars, const unsigned long int seed, const unsigned long int serial, const ABC::MPI_par* _mp
     ) const = 0;
 };
 
 // an AbcSimFun which throws an error if used. This is intended to be used as a default, so that if *not* replaced, the error
 // will be thrown when the simulator is used.
 struct AbcSimUnset : AbcSimFun {
-    vector<ABC::float_type> operator()(
-      vector<ABC::float_type> /*pars*/, const unsigned long int /*seed*/, const unsigned long int /*serial*/, const ABC::MPI_par* /*_mp*/
+    vector<float_type> operator()(
+      vector<float_type> pars, const unsigned long int seed, const unsigned long int serial, const ABC::MPI_par* _mp
```

Comment:
@pearsonca I think we still want the args commented:
```suggestion
      vector<float_type> /*pars*/, const unsigned long int /*seed*/, const unsigned long int /*serial*/, const ABC::MPI_par* /*_mp*/
```